pipeline/pipeline.go

```diff
@@ -366,12 +392,14 @@ func (p *Pipeline) finalize(event *Event, notifyInput bool, backEvent bool) {
 		p.totalCommitted.Inc()
 		p.totalSize.Add(int64(event.Size))
 
+		p.outSampleMu.Lock()
 		if len(p.outSample) == 0 && rand.Int()&1 == 1 {
```

Comment:
why rand thing is needed? can we leave a comment?

Reply:
I'll add comment, it's just way to say 50% possibility, like `rand.Int() % 2 == 1`

Reply:
I think we could remove 50% possibility as it was added for debug

Reply:
Done.